iroh/src/sync/live.rs

```diff
@@ -544,29 +586,52 @@ impl<S: store::Store> Actor<S> {
                 debug!(topic = ?topic, "broadcast new entry");
                 self.gossip.broadcast(topic, message).await?;
 
+                // Notify subscribers about the event
                 if let Some(subs) = subs {
-                    futures::future::join_all(subs.values().map(|sub| {
-                        sub(LiveEvent::InsertLocal {
-                            entry: entry.clone(),
-                        })
-                    }))
-                    .await;
+                    let event = LiveEvent::InsertLocal {
+                        entry: entry.clone(),
+                    };
+                    notify_all(subs, event).await;
                 }
             }
             InsertOrigin::Sync(peer_id) => {
+                let from = peer_id.and_then(|id| PeerId::from_bytes(&id).ok());
+                let entry = signed_entry.entry();
+                let hash = *entry.record().content_hash();
+
+                // A new entry was inserted from initial sync or gossip. Queue downloading the
+                // content.
+                let content_status = if let Some(_bao_entry) = self.bao_store.get(&hash) {
```

Comment:
`.is_some()` is a bit cleaner

Reply:
would also be nice, if there was a method for this on the store, eg `.contains`, that could look this up without having to move any memory

Reply:
Would leave that for a separate PR. Also I just noticed that partial entries are not properly covered.